### DISCUSSION TITLE
MacOS performance loss when updating to newer Air SDK

### DISCUSSION THREAD

**Discussion by chrslgr** [airsdk/Adobe-Runtime-Support#2197] [General]
hi. 

we are seeing a performance loss on our mac-app (not on other platforms) when updating air from 33.1.1.217 to .633, .713, .929 (BigSur 11.6.3, i7). (runtime is included in the build)

debug runs smooth as usual, only the release build is slow. every 5 seconds or so the app doesn't respond anymore for 3, 4 seconds. 

what could be the reason for that? could you give us a hint where or how to find the issue?

we would need to update since some of our customers have the fn-textinput-bug at lower e and d, which is mentioned [here](https://github.com/airsdk/Adobe-Runtime-Support/issues/1323).

thanks, chris

**Comment by ajwfrost** [comment]
That's an odd one ... are you able to reproduce it yourself? If so, can you get a Scout trace of it?
If you wanted to send us the .app bundle, we could try to reproduce here and could then replace the runtime with a version that has extra profiling/debugging in it...?

thanks

**Comment by chrslgr** [comment]
i haven't tried with scout, just downloaded it and am testing.
i just sent you an email with the link for download.
thanks

**Comment by chrslgr** [comment]
when taking a look at scout, there is quite a lot of garbage collection taking place when/before the app doesn't respond. before this garbage collection are exception errors concerning encrypted local store:
<img width="390" alt="image" src="https://user-images.githubusercontent.com/4034463/195303754-69b8d800-9820-4b9d-ac5f-0568fe432a82.png">

**Comment by chrslgr** [comment]
<img width="1186" alt="image" src="https://user-images.githubusercontent.com/4034463/195304224-2185785c-b209-4eed-a2b2-dabc5c8873f6.png">

**Reply by ajwfrost** [reply]
So that looks like within the commit event handler, it's doing a huge number of attempts to read the application's ELS data but failing in this .. which then results in a lot of data usage which is triggering the garbage collection. Repeatedly. Very odd! I can't see anything in the changelog between 33.1.1.217 and 33.1.1.633 that would immediately explain it, we'll have to take a look running it locally..

thanks

**Reply by chrslgr** [reply]
thank you!

**Reply by ajwfrost** [reply]
Quick update here -> we'll have to check but the "garbage collection" may either be a misnomer by Scout, or it may just be caused by what's happening here - which is that every call to ELS is resulting in a full parse of all the files in the application to validate the code signatures. Status is coming back which seems to be correct, but the value is not being cached (!) which means, next time it's called, the same thing happens...

So we should be able to fix the delays and the reports of GC...

**Reply by itlancer** [reply]
For me it fixed with AIR 50.0.1.2+.
Thanks!

**Reply by chrslgr** [reply]
well, i'm not sure it is fixed. i rebuilt with 50.1.1.1. and the startup of my app takes around 40 seconds, used by garbage collection:
![image](https://user-images.githubusercontent.com/4034463/205654451-c1cf3faf-8f76-4ff4-b792-ed34ee9b0171.png)

once that is done though, the app runs smooth again...

**Reply by ajwfrost** [reply]
Hi @chrslgr  - is that ADL, or a build "app bundle" running there? Am wondering if it makes a difference if you code-sign it..
If it's an app bundle, would it be possible to send this to us so we can review it again to work out why this isn't actually caching the value (my assumption is that there's still some exception being thrown which means it's repeatedly trying to do all the security stuff over and over..)
Upload link: https://transfer.harman.com/requests/x7Ajs2qDYxhtd7p4YNa68N

thanks

**Reply by chrslgr** [reply]
hi andrew. it's a .pkg. in debug everything is fine. uploaded the file at your link. thanks!

**Reply by ajwfrost** [reply]
Thanks -> can certainly reproduce it, am not seeing much wrong in the Scout log other than the massive delay in GC! so we'll have to take a deep dive into this..
